src/core/qgsmagneticmodel.h

```diff
@@ -0,0 +1,299 @@
+
+    /**
+     * Returns the maximum order of the components of the model.
+     *
+     * \throws QgsNotSupportedException if GeographicLib is not available
+     * \see degree()
+     */
+    int order() const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Evaluates the components of the geomagnetic field at a point.
+     *
+     * \param years the time (in fractional years)
+     * \param latitude latitude of the point (in decimal degrees)
+     * \param longitude longitude of the point (in decimal degrees)
+     * \param height height of the point above the ellipsoid (in meters)
+     * \param Bx the easterly component of the magnetic field (in nanotesla).
+     * \param By the northerly component of the magnetic field (in nanotesla)
+     * \param Bz the vertical (up) component of the magnetic field (in nanotesla)
+     *
+     * \returns TRUE if the components were successfully calculated
+     *
+     * \see getComponentsWithTimeDerivatives()
+     */
+    bool getComponents( double years, double latitude, double longitude, double height, double &Bx SIP_OUT, double &By SIP_OUT, double &Bz SIP_OUT ) const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Evaluates the components of the geomagnetic field at a point, and their time derivatives.
+     *
+     * \param years the time (in fractional years)
+     * \param latitude latitude of the point (in decimal degrees)
+     * \param longitude longitude of the point (in decimal degrees)
+     * \param height height of the point above the ellipsoid (in meters)
+     * \param Bx the easterly component of the magnetic field (in nanotesla).
+     * \param By the northerly component of the magnetic field (in nanotesla)
+     * \param Bz the vertical (up) component of the magnetic field (in nanotesla)
+     * \param Bxt the rate of change of Bx (nT/yr)
+     * \param Byt the rate of change of By (nT/yr)
+     * \param Bzt the rate of change of Bz (nT/yr)
+     *
+     * \returns TRUE if the components were successfully calculated
+     *
+     * \see getComponents()
+     */
+    bool getComponentsWithTimeDerivatives( double years, double latitude, double longitude, double height, double &Bx SIP_OUT, double &By SIP_OUT, double &Bz SIP_OUT, double &Bxt SIP_OUT, double &Byt SIP_OUT, double &Bzt SIP_OUT ) const SIP_THROW( QgsNotSupportedException );
+
+    /**
+     * Compute various quantities dependent on a magnetic field.
+     *
+     * \param Bx the easterly component of the magnetic field (in nanotesla)
+     * \param By the northerly component of the magnetic field (in nanotesla)
+     * \param Bz the vertical (up) component of the magnetic field (in nanotesla)
+     * \param H the horizontal magnetic field (in nanotesla)
+     * \param F the total magnetic field (in nanotesla)
+     * \param D the declination of the field (degrees east of north)
+     * \param I the inclination of the field (degrees down from horizontal)
```

Comment:
I guess the main use case is to return declination and inclination. Maybe you can add convenient methods for that?